_drafts/2019-01-10-moving-client-packages-to-monorepo.md

```diff
@@ -0,0 +1,82 @@
+
+**In a monorepo the whole workflow can be simplified like this:**
+
+1.  Create a branch from develop
+2.  Create the function
+3.  Build, test
+4.  Merge back when done
+
+So much easier and cleaner than before. No need to copy anything, anywhere. No need to release a dev alpha or a minor version. You can see your changes in all the packages once you modified something. It not only made our life simpler but it gave us a productivity boost as well. Let's see how the transition went.
+
+![Patrick](/img/posts/monorepo/patrick-meme.jpg 'Get all the packages we extracted from one repository and put them together in a new one')
+
+## Moving sensenet client packages to monorepo
+
+First thing first, we need a new reopsitory. We could also use one of the repos and put everything there, but it felt cleaner to start a new one. I have created a todo list in this [pull request](https://github.com/SenseNet/sn-client/pull/1) to track the progress and to see what is left. 
+
+We moved all the packages to the packages folder with lerna. _[Lerna](https://lernajs.io/) is a tool that optimizes the workflow around managing multi-package repositories with git and npm._ This way we could see the commit history in the new repository as well. We had to make sure that every package is able to build itslef. We faced some problems here. Not every package was configured correctly. Some was missing dependencies that were needed for it to be able to build.
+
+After that we run the tests to make sure we didn't break anything. Because the repositories were not consistent in their configs, it failed. We had to temporarily suspend the linting to see the test results. With that said we realized that we should make all repositories consistent. Fortunately there were no big errors.
+
+This is the list that we did to make the packages unified:
+  - Styling - Added .editorconfig, prettier and precommit hook with [lint-staged](https://github.com/okonet/lint-staged)
+  - Linting - Pointed the tslint.json files to the root tslint file and fixed the errors
+  - Build - Pointed the tsconfig.json files to the base tsconfig file
+  - Updated the package.json files to reflect the new changes (issue link, git url), removed common devdependencies
+  - Removed files that are not needed anymore in all the repositories (travis.yml, contributing.md, .vscode folder)
+
+When we finished with all this we had to make the ci work again, update the readmes, archive the old repositories and move all the issues to the new repository. The last step was to do a minor release from the packages. It was a piece of cake with the help of Lerna.
+
+### After all here are the benefits we gained:
+
+- **It is easier to coordinate changes across modules** - Because every package that depends on another linked together, you can see the changes in no time.
```

Comment:
```suggestion
- **It is easier to coordinate changes across modules** - Because every package that depends on another is linked together, you can see the changes in no time.
```